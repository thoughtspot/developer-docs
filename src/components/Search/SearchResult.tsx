import React from 'react';
import cx from 'classnames';
type SearchResultProps = {
    index: number;
    highlightedIndex: number;
    keyword: string;
    title: string;
    isKeywordNotFound?: boolean;
};

const SearchResult = (props: SearchResultProps) => {
    const searchResultContent = props.title + '...';
    return (
        <div
            className={cx(
                'list-search-item',
                { textContainer: !props.isKeywordNotFound },
                { active: props.index === props.highlightedIndex },
            )}
            data-testid="search-result"
        >
            <p
                className="title"
                dangerouslySetInnerHTML={{ __html: props.keyword }}
            ></p>
            <p
                className="footer"
                dangerouslySetInnerHTML={{ __html: searchResultContent }}
            ></p>
        </div>
    );
};

export default SearchResult;
