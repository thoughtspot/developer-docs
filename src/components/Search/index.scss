@import '../../assets/styles/variables.scss';

.searchWrapper {
    padding: 20px $padding-md 20px $padding-md;
    border-bottom: 1px var(--border-color) solid;
    position: fixed;
    width: 100%;
    background: var(--nav-bar-color);
    z-index: $z-index-search;
    display: flex;

    height: '300px';
}

.inClusterSite {
    top: 0;
}

.searchInputWrapper {
    position: relative;
    // width: calc(100% - 600px);
    display: flex;
    align-items: center;

    .searchInputContainer {
        display: flex;
        align-items: center;
        width: 100%;
    }

    input {
        width: 100%;
        border: 1px solid var(--border-color);
        border-radius: 6px;
        height: $height-search-input;
        padding-left: $padding-left-search-input;
        font-size: $font-size-normal;
        color: var(--primary-color);
        font-family: $font-family-doc;
        padding-bottom: 1px;
        background: var(--body-background-color);
    }

    input:focus {
        outline: 0;
    }

    input::placeholder {
        color: $darkgrey;
    }

    .searchIcon {
        position: relative;
        left: 30px;
        height: $icon-width;
        width: $icon-height;
    }

    .resultContainer {
        position: absolute;
        background-color: var(--body-background-color);
        width: calc(100% - 20px);
        max-height: $result-container-max-height;
        top: 38px;
        border-radius: 6px;
        border: 1px solid var(--border-color);
        box-shadow: $box-shadow-search-box;
        overflow: auto;
        left: 20px;
    }

    .result {
        border-bottom: 1px solid var(--border-color);
        display: inline-block;
        vertical-align: top;
        width: 100%;
        padding: 8px 0px;

        .title {
            margin: 2px $margin-search-result 0px $margin-search-result;
            line-height: $line-height-result-title;
            font-size: $font-size-normal;
            font-weight: $font-weight-normal;
            color: var(--primary-color);
        }

        .footer {
            margin: 5px $margin-search-result;
            font-weight: $font-weight-normal;
            font-size: $font-size-small;
            line-height: $line-height-result-footer;
            color: var(--primary-color);
        }
        &:hover {
            background: $text-container-background;
        }
        cursor: pointer;
    }

    .active {
        background: $text-container-background;
    }
}

@media screen and (min-width: $tablet-resolution-min) and (max-width: $tablet-resolution-max) {
    .searchInputWrapper {
        width: 50%;
    }
}

@media screen and (max-width: $mobile-resolution-max) {
    .searchWrapper {
        left: 0;
        padding: 20px 5px 10px 10px;
        margin-right: 5px;
        width: 100%;

        .searchInputWrapper {
            width: 75%;
        }
    }

    .visHidden {
        visibility: hidden;
    }
}
