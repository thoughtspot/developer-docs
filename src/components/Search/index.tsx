import React, { useEffect, useState, useRef } from 'react';
import { IconContext } from '@react-icons/all-files';
import { BiSearch } from '@react-icons/all-files/bi/BiSearch';
import './index.scss';
import { SearchQueryResult } from '../../interfaces/index';
import {
    ARROW_DOWN,
    ARROW_UP,
    ENTER,
} from '../../constants/keystrokeConstants';
import t from '../../utils/lang-utils';
import SearchResult from './SearchResult';
// import BackButton from '../BackButton';

type SearchProps = {
    options: SearchQueryResult[];
    leftNavOpen: boolean;
    keyword: string;
    isMaxMobileResolution: boolean;
    isDarkMode: boolean;
    optionSelected: (pageid: string, sectionId: string) => void;
    onChange: (e: React.FormEvent<HTMLInputElement>) => void;
    updateKeyword: Function;
    setDarkMode: Function;
    isPublicSiteOpen: boolean;
    leftNavWidth: Number;
};

const Search: React.FC<SearchProps> = (props) => {
    const node = useRef();
    const anchor = useRef();
    const optionListRef = useRef({});
    const [showSearchResult, updateShowSearchResult] = useState(false);
    const [highlightedIndex, setHighlightedIndex] = useState(0);

    useEffect(() => {
        if (props.options.length > 0 && props.keyword) {
            updateShowSearchResult(true);
        } else {
            updateShowSearchResult(false);
        }
    }, [props.keyword, props.options]);

    // This handles the mouse click events for suggestion list
    const handleClick = (event: Event) => {
        if (node?.current?.contains(event.target)) {
            return;
        }
        updateShowSearchResult(false);
    };
    const onSearchOptionSelected = () => {
        props.updateKeyword('');
        if (searchInput.current) {
            searchInput.current.blur();
        }
        updateShowSearchResult(false);
    };
    useEffect(() => {
        document.addEventListener('mousedown', handleClick);
        return () => {
            document.removeEventListener('mousedown', handleClick);
        };
    }, [node]);

    const onFocus = () => updateShowSearchResult(true);
    const searchInput = useRef<HTMLInputElement>();
    const onKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
        if (props.options.length === 0) return;

        const optionSize = props.options.length;

        switch (e.key) {
            case ARROW_UP:
                e.preventDefault();
                setHighlightedIndex(
                    (prev: number) => (prev - 1 + optionSize) % optionSize,
                );
                return;
            case ARROW_DOWN:
                e.preventDefault();
                setHighlightedIndex((prev: number) => (prev + 1) % optionSize);
                return;
            case ENTER:
                if (props.options[highlightedIndex].type === 'text') {
                    return;
                }
                if (props.options[highlightedIndex].type === 'html') {
                    if (anchor?.current) {
                        anchor.current.click();
                    }
                } else {
                    props.optionSelected(
                        props.options[highlightedIndex].pageid,
                        props.options[highlightedIndex].sectionId,
                    );
                }
                onSearchOptionSelected();
                setHighlightedIndex(0);
                return;
            default:
                return;
        }
    };

    const renderOption = (option: SearchQueryResult, index: number) => {
        switch (option.type) {
            case 'html':
                return (
                    <a
                        data-testid="result-link"
                        key={option.pageid}
                        className="result"
                        href={option.link}
                        target="_blank"
                        ref={anchor}
                        onClick={onSearchOptionSelected}
                    >
                        <SearchResult
                            highlightedIndex={highlightedIndex}
                            index={index}
                            keyword={option.title}
                            title={
                                (option._snippetResult &&
                                    option._snippetResult?.body?.value) ??
                                ''
                            }
                        />
                    </a>
                );
            case 'text':
                return (
                    <div
                        key={option.pageid}
                        className="result"
                        ref={(el: HTMLDivElement) => {
                            optionListRef.current[index] = el;
                        }}
                    >
                        <SearchResult
                            highlightedIndex={highlightedIndex}
                            index={index}
                            keyword={props.keyword}
                            title={
                                (option._snippetResult &&
                                    option._snippetResult?.body?.value) ??
                                ''
                            }
                            isKeywordNotFound={true}
                        />
                    </div>
                );
            default:
                return (
                    <div
                        key={option.pageid}
                        className="result"
                        onClick={() => {
                            props.optionSelected(
                                option.pageid,
                                option.sectionId,
                            );
                            onSearchOptionSelected();
                        }}
                        ref={(el: HTMLDivElement) => {
                            optionListRef.current[index] = el;
                        }}
                    >
                        <SearchResult
                            highlightedIndex={highlightedIndex}
                            index={index}
                            keyword={`${option.title} | ${option.sectionTitle}`}
                            title={
                                (option._snippetResult &&
                                    option._snippetResult?.body?.value) ??
                                ''
                            }
                        />
                    </div>
                );
        }
    };

    if (optionListRef?.current[highlightedIndex]) {
        optionListRef?.current[highlightedIndex].scrollIntoView();
    }

    return (
        <div
        // className={`searchWrapper ${props.leftNavOpen ? 'visHidden' : ''} ${
        //     !props.isPublicSiteOpen ? 'inClusterSite' : ''
        // }`}
        >
            <div className="searchInputWrapper">
                <div className="searchInputContainer">
                    <IconContext.Provider
                        value={{
                            className: `icon searchIcon`,
                        }}
                    >
                        <BiSearch />
                    </IconContext.Provider>
                    <input
                        ref={searchInput}
                        data-testid="search-input"
                        type="Search"
                        placeholder={t('SEARCH_PLACEHOLDER')}
                        onFocus={onFocus}
                        onKeyDown={onKeyDown}
                        value={props.keyword}
                        onChange={props.onChange}
                        autoFocus
                    />
                </div>
                {showSearchResult && props.options?.length ? (
                    <div
                        ref={node}
                        className="resultContainer"
                        data-testid="resultContainer"
                    >
                        {props.options.map(
                            (option: SearchQueryResult, index: number) => {
                                return renderOption(option, index);
                            },
                        )}
                    </div>
                ) : (
                    ''
                )}
            </div>
        </div>
    );
};

export default Search;
