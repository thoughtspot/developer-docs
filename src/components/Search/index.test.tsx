import React from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react';
import { act } from 'react-dom/test-utils';

import Search from './index';
import { SearchQueryResult } from '../../interfaces/index';

jest.mock('../../utils/lang-utils', () => (key) => key);
const getPageUrl = () => window.location.href.split(/[?#]/)[0];
window.HTMLElement.prototype.scrollIntoView = jest.fn();

describe('Search', () => {
    const searchPropsWithOptions = {
        keyword: 'test',
        onChange: jest.fn(),
        options: [
            {
                pageid: 'page1',
                title: 'test title',
                _snippetResult: {
                    body: {
                        value: 'test content',
                    },
                },
                type: 'html',
                link: '/',
            },
            {
                pageid: 'page2',
                title: 'test title',
                _snippetResult: {
                    body: {
                        value: 'test content',
                    },
                },
                type: 'html',
                link: '/',
            },
            {
                pageid: 'page3',
                title: 'test title',
                _snippetResult: {
                    body: {
                        value: 'test content',
                    },
                },
                type: 'text',
                link: '/',
            },
        ] as SearchQueryResult[],
        optionSelected: jest.fn(),
        leftNavOpen: false,
        updateKeyword: jest.fn(),
        isMaxMobileResolution: false,
        setDarkMode: jest.fn(),
        isDarkMode: false,
        isPublicSiteOpen: false,
    };

    const searchPropsWithoutOptions = {
        keyword: 'test keyword',
        onChange: jest.fn(),
        options: [] as SearchQueryResult[],
        optionSelected: jest.fn(),
        leftNavOpen: true,
        updateKeyword: jest.fn(),
        isMaxMobileResolution: false,
        setDarkMode: jest.fn(),
        isDarkMode: false,
        isPublicSiteOpen: false,
    };

    it('should render correctly without options', () => {
        const { container, queryByTestId } = render(
            <Search {...searchPropsWithoutOptions} />,
        );
        // expect(container).toMatchSnapshot();
        expect(queryByTestId('search-result')).toBeNull();
        expect(queryByTestId('resultContainer')).toBeNull();
    });

    it('should not show search results when options are not present and keyword is blank', () => {
        const { queryByTestId } = render(
            <Search {...searchPropsWithoutOptions} />,
        );
        expect(queryByTestId('search-result')).toBeNull();
        expect(queryByTestId('resultContainer')).toBeNull();
    });

    it('should render correctly with options', async () => {
        const {
            container,
            queryByTestId,
            queryAllByTestId,
            getByTestId,
        } = render(<Search {...searchPropsWithOptions} />);
        const input = await getByTestId('search-input');
        const resultContainer = await getByTestId('resultContainer');

        act(() => {
            fireEvent.focus(input);
        });

        // expect(container).toMatchSnapshot();
        expect(queryAllByTestId('search-result').length).toBe(3);
        expect(resultContainer).toBeInTheDocument();
    });

    it('should show results on focus', async () => {
        const { queryByTestId, queryAllByTestId, getByTestId } = render(
            <Search {...searchPropsWithOptions} />,
        );
        const input = await getByTestId('search-input');
        const resultContainer = await getByTestId('resultContainer');

        act(() => {
            fireEvent.focus(input);
        });
        expect(queryAllByTestId('search-result').length).toBe(3);
        expect(resultContainer).toBeInTheDocument();
    });

    it('should not show results on mouse down input', async () => {
        const { queryByTestId, queryAllByTestId, getByTestId } = render(
            <Search {...searchPropsWithOptions} />,
        );
        const input = await getByTestId('search-input');
        const resultContainer = await getByTestId('resultContainer');

        act(() => {
            fireEvent.mouseDown(input);
        });
        expect(queryAllByTestId('search-result').length).toBe(0);
        expect(resultContainer).not.toBeInTheDocument();
    });

    it('should highlight text on key down arrow down and arrow up ', async () => {
        const { container, queryAllByTestId, getByTestId } = render(
            <Search {...searchPropsWithOptions} />,
        );
        const input = await getByTestId('search-input');

        act(() => {
            fireEvent.focus(input);
            fireEvent.keyDown(input, { key: 'ArrowDown', code: 'ArrowDown' });
        });

        // expect(container).toMatchSnapshot();
        expect(queryAllByTestId('search-result')[1]).toHaveClass('active');

        act(() => {
            fireEvent.keyDown(input, { key: 'ArrowUp', code: 'ArrowUp' });
        });
        // expect(queryAllByTestId('search-result')[0]).toHaveClass('active');
        expect(
            queryAllByTestId('search-result')[0].firstChild.textContent,
        ).toBe('test title');
        expect(input.value).toBe('test');
    });

    it('should keyword and input value on enter', async () => {
        const { container, queryAllByTestId, getByTestId } = render(
            <Search {...searchPropsWithOptions} />,
        );
        const input = await getByTestId('search-input');

        act(() => {
            fireEvent.keyDown(input, { key: 'Enter', code: 'Enter' });
        });

        expect(queryAllByTestId('search-result').length).toBe(0);
        //hiding search suggestions once the user clicks enter
    });

    it('should do nothing if any key other than arrow up, down and enter is pressed', async () => {
        const { container, queryAllByTestId, getByTestId } = render(
            <Search {...searchPropsWithOptions} />,
        );
        const input = await getByTestId('search-input');

        act(() => {
            fireEvent.keyDown(input, { key: 'ArrowRight', code: 'ArrowRight' });
        });

        expect(queryAllByTestId('search-result')[1]).not.toHaveClass('active');
    });

    it('should do nothing if option type is text', async () => {
        const { container, queryAllByTestId, getByTestId } = render(
            <Search {...searchPropsWithOptions} />,
        );
        const input = await getByTestId('search-input');

        act(() => {
            fireEvent.keyDown(input, { key: 'ArrowDown', code: 'ArrowDown' });
            fireEvent.keyDown(input, { key: 'ArrowDown', code: 'ArrowDown' });
        });

        expect(queryAllByTestId('search-result')[2]).toHaveClass('active');
    });
});
