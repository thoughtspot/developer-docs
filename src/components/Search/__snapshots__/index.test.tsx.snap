// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should highlight text on key down arrow down and arrow up  1`] = `
<div>
  <div
    class="searchWrapper  inClusterSite"
  >
    <div>
      <div
        class="backButtonWrapper"
        data-testid="backBtn"
        style="padding-left: 0px;"
      >
        <button>
          <a
            target="_parent"
          >
            <svg
              class="icon leftIcon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M5.854 4.646a.5.5 0 010 .708L3.207 8l2.647 2.646a.5.5 0 01-.708.708l-3-3a.5.5 0 010-.708l3-3a.5.5 0 01.708 0z"
                fill-rule="evenodd"
              />
              <path
                clip-rule="evenodd"
                d="M2.5 8a.5.5 0 01.5-.5h10.5a.5.5 0 010 1H3a.5.5 0 01-.5-.5z"
                fill-rule="evenodd"
              />
            </svg>
          </a>
        </button>
        <p>
          NAV_BACK_BTN_TEXT
        </p>
      </div>
    </div>
    <div
      class="searchInputWrapper"
    >
      <div
        class="searchInputContainer"
      >
        <svg
          class="icon searchIcon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10,18c1.846,0,3.543-0.635,4.897-1.688l4.396,4.396l1.414-1.414l-4.396-4.396C17.365,13.543,18,11.846,18,10 c0-4.411-3.589-8-8-8s-8,3.589-8,8S5.589,18,10,18z M10,4c3.309,0,6,2.691,6,6s-2.691,6-6,6s-6-2.691-6-6S6.691,4,10,4z"
          />
        </svg>
        <input
          data-testid="search-input"
          placeholder="SEARCH_PLACEHOLDER"
          type="Search"
          value="test"
        />
      </div>
      <div
        class="resultContainer"
        data-testid="resultContainer"
      >
        <a
          class="result"
          data-testid="result-link"
          href="/"
          target="_blank"
        >
          <div
            class="textContainer"
            data-testid="search-result"
          >
            <p
              class="title"
            >
              test title
            </p>
            <p
              class="footer"
            >
              test content...
            </p>
          </div>
        </a>
        <a
          class="result"
          data-testid="result-link"
          href="/"
          target="_blank"
        >
          <div
            class="textContainer active"
            data-testid="search-result"
          >
            <p
              class="title"
            >
              test title
            </p>
            <p
              class="footer"
            >
              test content...
            </p>
          </div>
        </a>
        <div
          class="result"
        >
          <div
            class=""
            data-testid="search-result"
          >
            <p
              class="title"
            >
              test
            </p>
            <p
              class="footer"
            >
              test content...
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Search should render correctly with options 1`] = `
<div>
  <div
    class="searchWrapper  inClusterSite"
  >
    <div>
      <div
        class="backButtonWrapper"
        data-testid="backBtn"
        style="padding-left: 0px;"
      >
        <button>
          <a
            target="_parent"
          >
            <svg
              class="icon leftIcon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M5.854 4.646a.5.5 0 010 .708L3.207 8l2.647 2.646a.5.5 0 01-.708.708l-3-3a.5.5 0 010-.708l3-3a.5.5 0 01.708 0z"
                fill-rule="evenodd"
              />
              <path
                clip-rule="evenodd"
                d="M2.5 8a.5.5 0 01.5-.5h10.5a.5.5 0 010 1H3a.5.5 0 01-.5-.5z"
                fill-rule="evenodd"
              />
            </svg>
          </a>
        </button>
        <p>
          NAV_BACK_BTN_TEXT
        </p>
      </div>
    </div>
    <div
      class="searchInputWrapper"
    >
      <div
        class="searchInputContainer"
      >
        <svg
          class="icon searchIcon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10,18c1.846,0,3.543-0.635,4.897-1.688l4.396,4.396l1.414-1.414l-4.396-4.396C17.365,13.543,18,11.846,18,10 c0-4.411-3.589-8-8-8s-8,3.589-8,8S5.589,18,10,18z M10,4c3.309,0,6,2.691,6,6s-2.691,6-6,6s-6-2.691-6-6S6.691,4,10,4z"
          />
        </svg>
        <input
          data-testid="search-input"
          placeholder="SEARCH_PLACEHOLDER"
          type="Search"
          value="test"
        />
      </div>
      <div
        class="resultContainer"
        data-testid="resultContainer"
      >
        <a
          class="result"
          data-testid="result-link"
          href="/"
          target="_blank"
        >
          <div
            class="textContainer active"
            data-testid="search-result"
          >
            <p
              class="title"
            >
              test title
            </p>
            <p
              class="footer"
            >
              test content...
            </p>
          </div>
        </a>
        <a
          class="result"
          data-testid="result-link"
          href="/"
          target="_blank"
        >
          <div
            class="textContainer"
            data-testid="search-result"
          >
            <p
              class="title"
            >
              test title
            </p>
            <p
              class="footer"
            >
              test content...
            </p>
          </div>
        </a>
        <div
          class="result"
        >
          <div
            class=""
            data-testid="search-result"
          >
            <p
              class="title"
            >
              test
            </p>
            <p
              class="footer"
            >
              test content...
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Search should render correctly without options 1`] = `
<div>
  <div
    class="searchWrapper visHidden inClusterSite"
  >
    <div>
      <div
        class="backButtonWrapper"
        data-testid="backBtn"
        style="padding-left: 0px;"
      >
        <button>
          <a
            target="_parent"
          >
            <svg
              class="icon leftIcon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M5.854 4.646a.5.5 0 010 .708L3.207 8l2.647 2.646a.5.5 0 01-.708.708l-3-3a.5.5 0 010-.708l3-3a.5.5 0 01.708 0z"
                fill-rule="evenodd"
              />
              <path
                clip-rule="evenodd"
                d="M2.5 8a.5.5 0 01.5-.5h10.5a.5.5 0 010 1H3a.5.5 0 01-.5-.5z"
                fill-rule="evenodd"
              />
            </svg>
          </a>
        </button>
        <p>
          NAV_BACK_BTN_TEXT
        </p>
      </div>
    </div>
    <div
      class="searchInputWrapper"
    >
      <div
        class="searchInputContainer"
      >
        <svg
          class="icon searchIcon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10,18c1.846,0,3.543-0.635,4.897-1.688l4.396,4.396l1.414-1.414l-4.396-4.396C17.365,13.543,18,11.846,18,10 c0-4.411-3.589-8-8-8s-8,3.589-8,8S5.589,18,10,18z M10,4c3.309,0,6,2.691,6,6s-2.691,6-6,6s-6-2.691-6-6S6.691,4,10,4z"
          />
        </svg>
        <input
          data-testid="search-input"
          placeholder="SEARCH_PLACEHOLDER"
          type="Search"
          value="test keyword"
        />
      </div>
      
    </div>
  </div>
</div>
`;
